stoservice/src/main/java/com/tutrit/stoservice/repository/EngineerRepository.java

```diff
@@ -0,0 +1,101 @@
+package com.tutrit.stoservice.repository;
+
+import com.tutrit.stoservice.bean.Engineer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+
+public class EngineerRepository {
+    Engineer ext;
+    Engineer extFind;
+    Engineer[] engineers = new Engineer[3];
+    Logger log = LoggerFactory.getLogger(EngineerRepository.class);
+
+    public Engineer createEngineer(Engineer engineer) {
+        int ret = 0;
+        for (int i = 0; i < engineers.length; i++) {
+            if (engineers[i] == null) {
+                engineer.setIdEngineer(count(i));
+                ret = i;
+                engineers[i] = engineer;
+                if ((i * 2) >= engineers.length) {
+                    Engineer[] engineersTemp = Arrays.copyOf(engineers, (engineers.length + 5));
+                    engineers = engineersTemp;
+                }
+                break;
+            }
+        }
+        return engineers[ret];
+    }
+
+    public Engineer findEngineer(int i) {
+        if (i > (engineers.length - 1)) {
+            log.info("Array out of bounds");
+            return new Engineer();
+        }
+        return engineers[i - 1];
+    }
+
+    public Engineer findEngineer(String a) {
+        int i = Integer.parseInt(a);
+        return findEngineer(i);
+    }
+
+    public Engineer findEngineer(Engineer engineer) {
+        Engineer find = extract(engineer);
+        for (Engineer e : engineers) {
+            extFind = extract(e);
+            if (find.equals(extFind)) {
```

Comment:
What is the need to extract the engineer if you can compare it directly in the if statement?